libcxx/src/CMakeLists.txt

```diff
@@ -397,6 +397,9 @@ if (NOT CMAKE_CONFIGURATION_TYPES)
     if(LIBCXX_INSTALL_HEADERS)
```

Comment:
In your commit message, please explain what the patch does. In particular, please explain the location where we install the `.cppm` files and the manifest file, and mention that the manifest file is supposed to live alongside the `dylib`, and that there's a contract with Clang. Also mention the command-line that can be used to find the manifest file from `clang`.

You referenced the SG15 discussion, but I would like us to capture the important points of that discussion in the commit message.